funnel/models/event.py

```diff
@@ -146,9 +148,18 @@ def remove_events(self, events):
 
     @classmethod
     def checkin_list(cls, event):
-        participant_attendee_join = db.join(Participant, Attendee, Participant.id == Attendee.participant_id)
-        stmt = db.select([Participant.id, Participant.fullname, Participant.email, Participant.company, Participant.twitter, Participant.puk, Participant.key, Attendee.checked_in, Participant.badge_printed]).select_from(participant_attendee_join).where(Attendee.event_id == event.id).order_by(Participant.fullname)
-        return db.session.execute(stmt).fetchall()
+        """
+        Returns participant details along with their associated ticket types as a comma-separated string.
+        WARNING: This query uses `string_agg` and hence will only work in PostgreSQL >= 9.0
+        """
+        participant_list = db.session.query('id', 'fullname', 'email', 'company', 'twitter', 'puk', 'key', 'checked_in', 'badge_printed', 'ticket_type_titles').from_statement(text('''
+            SELECT distinct(participant.id), participant.fullname, participant.email, participant.company, participant.twitter, participant.puk, participant.key, attendee.checked_in, participant.badge_printed,
+            (select string_agg(title, ',') from sync_ticket INNER JOIN ticket_type ON sync_ticket.ticket_type_id = ticket_type.id where sync_ticket.participant_id = participant.id) AS ticket_type_titles
+            FROM participant INNER JOIN attendee ON participant.id = attendee.participant_id LEFT OUTER JOIN sync_ticket ON participant.id = sync_ticket.participant_id
+            WHERE attendee.event_id = {event_id}
+            ORDER BY participant.fullname
+        '''.format(event_id=event.id))).all()
```

Comment:
Don't use `.format` on SQL strings. You're risking a SQL injection attack. Do this instead:

1. Use ":parameters" in the text: `WHERE attendee.event_id = :event_id`
2. `from_statement` returns an object that accepts parameters: `db.session.query.from_statement(db.text('...')).params(event_id=event.id).all()`